Update to new scalismo version

Andreas-Forster opened the pull request:
An initial preview of changes needed to update to new scalismo dependency and the complete migrating to scala 3. Yet not all warnings are fixed.

Andreas-Forster committed:
update to new scalismo version

Andreas-Forster committed:
added missing build.sbt

marcelluethi commented on src/main/scala/scalismo/faces/sampling/face/proposals/SphericalHarmonicsLightProposals.scala:
```diff
@@ -251,7 +250,7 @@ object SphericalHarmonicsLightProposals {
             totProb += 3.0 * (-0.5 * math.pow(diff.x / sdev, 2) - 0.5 * math.log(2.0 * math.Pi) - math.log(sdev))
           } else {
-            return Double.NegativeInfinity //if non monochromatic change
+            returning(Double.NegativeInfinity) //if non monochromatic change
```
what does returning mean? Shouldn't this just be `Double.NegativeInfinity`?

Andreas-Forster replied:
There was a warning that returning from an nested anonymous function is deprecated since Scala 3. And as far as I understood, the returning is a possible library replacement from scala.util.control.NonLocalReturns.

marcelluethi commented on src/test/scala/scalismo/faces/momo/MoMoTests.scala:
```diff
@@ -162,7 +167,7 @@ class MoMoTests extends FacesTestSuite {
     }
 
-    val f = File.createTempFile("momo", ".h5")
+    val f = File.createTempFile("momo", ".json")
```
I usually call the files `.h5.json` but this works as well.

Andreas-Forster replied:
That is a good idea - thx.

marcelluethi commented on project/build.properties:
```diff
@@ -1 +1 @@
-sbt.version=1.3.5
+sbt.version=1.6.1
```
Maybe we should go directly to 1.8?

Andreas-Forster replied:
I don't mind - will update that.

marcelluethi reviewed (commented):
Thanks for starting this update. This is great. 
I added a few comments inline. 

What surprises me is that I cannot see a change in build.sbt?

marcelluethi commented:
What happened to the integration tests? I see that no CI is running?

Andreas-Forster commented:
You were right, the build.sbt was indeed missing. I corrected that now.

I do not yet know what happened to the CI. Maybe never setup correctly or it was not maintained and is now outdated.

Andreas-Forster committed:
added CI, scalafmt and minor cleanup

Andreas-Forster committed:
fix missing library in CI

Andreas-Forster committed:
fixed formatting in sbt and tests

Andreas-Forster commented:
I added CI and fixed formatting to follow the style checks already used in Scalismo.

Further, I updated the file endings of models to be ".h5.json" in the tests and it uses now sbt 1.8.2 for building.

marcelluethi commented on build.sbt:
```diff
@@ -15,28 +15,26 @@ lazy val root = (project in file("."))
-      "org.scalatest" %% "scalatest" % "3.0.8" % "test"
+      "ch.unibas.cs.gravis" %% "scalismo" % "0.92-SNAPSHOT",
+      "org.scalatest" %% "scalatest" % "3.2.15" % Test,
+      "io.spray" %%  "spray-json" % "1.3.6"
```
Maybe we should think about replacing spray json with ujson. We use it only in two files and in scalismo we migrated to it recently. The less dependencies we are introducing the better. 
But it any case, this should be a separate PR, which I also would be willing to do.

marcelluethi reviewed (approved):
Thanks a lot for all the changes. This looks very good.

marcelluethi commented on build.sbt:
```diff
@@ -6,37 +6,40 @@ val scalismoPlatform = {
+    scalacOptions ++= Seq("-deprecation", "-unchecked", "-feature"), // "-opt:l:method",
+    javacOptions ++= Seq("-source", "1.8", "-target", "1.8"),
+    libraryDependencies ++= Seq(
+      "ch.unibas.cs.gravis" %% "scalismo" % "0.92-SNAPSHOT",
```
This can now be replaced with 0.92-RC1.

Andreas-Forster committed:
updated to RC1 of scalismo
